src/gap.c

```diff
@@ -291,7 +290,7 @@ static Obj Shell(Obj    context,
 
   }
   
-  STATE(PrintObjDepth) = oldPrintDepth;
+  SetPrintObjState(oldPrintDepth);
```

Comment:
Now that I pushed this, I see that there is yet another bug lurking here: we reset the print stack *depth*, but not its contents. So that could potentially lead to issues, and even crashes, as the objects "on" that stack could be changed now.

To fix this, I think we'd have to save and restore the whole printobj stack with all its content. Ugh.

My next thought was, "OK, let's just set the depth to 0 here, then while resulting in a botched printing (the one we interrupted by a break loop), otherwise we should be safe". But that's of course not true, as the kernel C functions `ViewObj` and `PrintObj` unconditionally decrement `PrintObjDepth` after the printing. So we'd end up with an underflow. OK, so next we can of course insert a check for that: if `PrintObjDepth == 0` before the point were we decrement it, then immediately abort the printing.

That's probably the quickest way to "fix" that issue. For a proper fix, I think we should just move the printobj state into a GAP back (or a bunch of them.. perhaps a bunch of plists), then it'll be easy to store and reset them when entering / leaving `Shell`.